tapgarden/planter.go

```diff
@@ -676,13 +684,42 @@ func (c *ChainPlanter) gardener() {
 }
 
 // finalizeBatch creates a new caretaker for the batch and starts it.
-func (c *ChainPlanter) finalizeBatch(
-	feeRate *chainfee.SatPerKWeight) (*BatchCaretaker, error) {
+func (c *ChainPlanter) finalizeBatch(params *FinalizeParams) (*BatchCaretaker,
+	error) {
+
+	var (
+		feeRate  *chainfee.SatPerKWeight
+		rootHash *chainhash.Hash
+		err      error
+	)
+
+	ctx, cancel := c.WithCtxQuit()
+	// If a tapscript tree was specified for this batch, we'll store it on
+	// disk. The caretaker we start for this batch will use it when deriving
+	// the final Taproot output key.
+	// TODO(jhb): Make this load call mandatory once we have a DB-backed
+	// implementation of the tapscript tree store.
+	if params != nil {
+		params.FeeRate.WhenSome(func(fr chainfee.SatPerKWeight) {
+			feeRate = &fr
+		})
+		params.TapTree.WhenSome(func(p asset.TapscriptTreePreimage) {
+			if c.cfg.TreeStore != nil {
```

Comment:
Ah I see what you mean here re that extra nil check at times. In this case, I think it'll go away once we have a proper DB version.

Reply:
Params could similarly be `fn.Option`? Nesting would be the same.

Reply:
I think we'd have to switch all the public funcs. to be `fn.Option`. We could use an option as the return from `typedParam` and get rid of one level of nesting though.